src/store/ducks/opensrp/hierarchies/index.tsx

```diff
@@ -623,14 +569,24 @@ export const getSelectedHierarchy = () => {
 
     // create an object with uniq jurisdiction entries for all jurisdictions that exist
     // in a path that has a selected leaf node.
-    allSelectedLeafNodes.forEach(node => {
-      allNodesInPaths = {
-        ...allNodesInPaths,
-        ...keyBy<TreeNode>(node.getPath(), nd => nd.model.id),
-      };
-    });
+    const temp = [];
+    // tslint:disable-next-line: prefer-for-of
+    for (let i = 0; i < allSelectedLeafNodes.length; i++) {
+      const thisNode = allSelectedLeafNodes[i];
+      const nodesInPath = thisNode.getPath();
+
+      // tslint:disable-next-line: prefer-for-of
+      for (let idx = 0; idx < nodesInPath.length; idx++) {
+        temp.push(nodesInPath[idx]);
+      }
+    }
```

Comment:
Is it at all possible to avoid the nested loops?

Reply:
yeah, Ill refactor that.

Reply:
@moshthepitt unfortunately I was not able to get around this. IMO I cannot avoid them here.